zScanner/Views/Departments/DepartmentButton.swift

```diff
@@ -29,6 +29,12 @@ class DepartmentButton: PrimaryButton {
     override var isSelected: Bool {
         didSet {
             backgroundColor = isSelected ? .primaryDark : .primary
+            
+            if isSelected {
```

Comment:
Activity indicator have to be connected to the viewModel state, not to the selection of the button. 

We always need to react to the origin of the event, not to the side-effect. The side effect may change and the loading might break.